hypershift-operator/controllers/hostedcluster/hostedcluster_controller.go

```diff
@@ -1498,6 +1498,19 @@ func (r *HostedClusterReconciler) delete(ctx context.Context, req ctrl.Request,
 	if err := r.Delete(ctx, ns); err != nil && !apierrors.IsNotFound(err) {
 		return false, fmt.Errorf("failed to delete namespace: %w", err)
 	}
+
+	secretSuffixesToRemove := []string{"pull-secret", "provider-creds", "signing-key", "ssh-key", "admin-kubeconfig"}
```

Comment:
Other than the generated `admin-kubeconfig` these resources are not managed by `hostedcluster_controller`, they need to be created ahead of time and referenced by the hostedCluster API, it just happen that our dev cli automate their creation for convenience. So it is expected for them to be also deleted out of band. Do you agree?

Reply:
For the `admin-kubeconfig` we can possibly enforce an `owner reference` during creation so it just gets garbage collected once it goes orphan when deleting the `hostedCluster`

Reply:
I'm totally cool with that! Just wasn't sure what the general strategy was there!

Reply:
I'll close since that makes sense